### PR TITLE
Add files via upload

### LINKED ISSUE #22: Implement file browser widget for each environment variable input

### PR DESCRIPTION
**Pull request by t-arnold**
implemented file browser widget for each environmental input in the submit stage. closes #22

### CODE REVIEW

File: uit/gui_tools/submit.py [hunk 1]
```diff
@@ -50,8 +51,10 @@ def pbs_options_view(self):
 class PbsScriptAdvancedInputs(HpcConfigurable):
     env_names = param.List()
     env_values = param.List()
+    env_browser = param.List()
     browse = param.Action(lambda self: self.toggle_file_browser(), label='📂')
     browse_toggle = param.Boolean(default=False)
```
Comment (sdc50) [inline uit/gui_tools/submit.py]: I believe we can remove both the `browse` and the `browse_toggle` parameters now, right? Are they being used anywhere?

File: uit/gui_tools/submit.py [hunk 2]
```diff
@@ -75,8 +78,24 @@ def env_var_widget(self, val, tag, **kwargs):
...
+            new_browser_pn = HpcFileBrowser(self.uit_client)
+            new_browser_pn.callback = partial(self.new_callback, a=new_val, b=new_browser_pn)
+            self.browse_view[:] = [new_browser_pn.panel]
+
+    @staticmethod
+    def new_callback(new_selection, a, b):
+        a.value = b.value[0]
 
     @param.depends('browse_toggle')
     def file_browser_view(self):
```
Comment (sdc50) [inline uit/gui_tools/submit.py]: This method is no longer needed and can be removed.

File: uit/gui_tools/submit.py [hunk 3]
```diff
@@ -89,28 +108,31 @@ def file_browser_view(self):
...
             name_widget = self.env_var_widget(val=k, tag=f'env_key_{i}')
             val_widget = self.env_var_widget(val=str(v), tag=f'env_val_{i}')
+            browser_widget = self.env_file_browser_widget(tag=f'env_browser_{i}')
             self.env_names.append(name_widget)
             self.env_values.append(val_widget)
+            self.env_browser.append(browser_widget)
 
         self.env_names.append(self.env_var_widget(val=None, tag='env_key_-1', placeholder='NEW_ENV_VAR'))
-        self.env_values.append(self.env_var_widget(val=None, tag='env_val_-1', disabled=True))
+        self.env_values.append(self.env_var_widget(val=None, tag='env_val_-1', placeholder=''))
```
Comment (sdc50) [inline uit/gui_tools/submit.py:113]: The reason for this being disabled was so a user couldn't edit the value field without there being a key. Does this make that possible?

File: uit/gui_tools/submit.py [hunk 4]
```diff
@@ -89,28 +108,31 @@ def file_browser_view(self):
...
+        self.env_browser.append(self.env_file_browser_widget(tag=f'env_browser_-1'))
```
[the remark this answers is not in the document]
Comment (sdc50) [reply uit/gui_tools/submit.py]:
```suggestion
        self.env_browser.append(self.env_file_browser_widget(tag='env_browser_-1'))
```

**Review by sdc50 — CHANGES_REQUESTED** [review]
@the-thomas-arnold this looks great. There's just a little bit of cleanup to do of code that is now obsolete. Let me know when that is done and we'll get this merged.

File: uit/gui_tools/submit.py (hunk 3 above)
Comment (t-arnold) [reply uit/gui_tools/submit.py:113]: Unfortunately, adding ```disabled=True``` did not disable the display of that single file browser button for the first row.

File: uit/gui_tools/submit.py (hunk 1 above)
Comment (t-arnold) [reply uit/gui_tools/submit.py]: That is correct, and I have removed them in the newest version of the file.

File: uit/gui_tools/submit.py (hunk 2 above)
Comment (t-arnold) [reply uit/gui_tools/submit.py]: Removed, thank you.

**Review by sdc50 — APPROVED** [review]

**Pull request merged**